docs: Mention middleware for getStaticProps (#33273)
vercel/next.js: docs/basic-features/data-fetching/get-static-props.md

- The data can be publicly cached (not user-specific)
- The page must be pre-rendered (for SEO) and be very fast — `getStaticProps` generates `HTML` and `JSON` files, both of which can be cached by a CDN for performance

> When combined with [Incremental Static Regeneration](/docs/basic-features/data-fetching/incremental-static-regeneration.md), it will run in the background while the stale page is being revalidated, and the fresh page served to the browser.
> 
> Because `getStaticProps` runs at build time, it does **not** receive data that’s only available during request time, such as query parameters or `HTTP` headers, as it generates static `HTML`. If you need access to the request for your page, e.g. for authentication purposes, consider using [middlewares](/docs/middleware.md) additionally to `getStaticProps`.

Reviewer: ```suggestion
When combined with [Incremental Static Regeneration](/docs/basic-features/data-fetching/incremental-static-regeneration.md), `getStaticProps` will run in the background while the stale page is being revalidated, and the fresh page served to the browser.

Because `getStaticProps` runs at build time, it does **not** have access to the incoming request (such as query parameters or `HTTP` headers) as it generates static `HTML`. If you need access to the request for your page, consider using [Middleware](/docs/middleware.md) in addition to `getStaticProps`.
```
